File: tutorial/worked-examples/bac-replication-r/analysis/code/01_summary_stats.R
# 01_summary_stats.R - Summary statistics

library(dplyr)
library(tidyr)

# Load analysis data
analysis_data <- readRDS(file.path(build, "output", "analysis_data.rds"))

# Core variables always present
vars <- c("total_fatalities", "hr_fatalities", "nhr_fatalities",
          "ln_hr", "ln_nhr", "treated")
labels <- c("Total Fatalities", "HR Fatalities", "Non-HR Fatalities",
            "Log HR Fatalities", "Log Non-HR Fatalities", "Treated")

# Add control variables if available
if ("unemployment" %in% names(analysis_data)) {
  vars <- c(vars, "unemployment")
  labels <- c(labels, "Unemployment")
}
if ("income" %in% names(analysis_data)) {
  vars <- c(vars, "income")
  labels <- c(labels, "Income")
}

# Summary statistics for main variables
summary_stats <- analysis_data %>%
  summarise(
    n_obs = n(),
    across(all_of(vars),
           list(
             mean = ~mean(., na.rm = TRUE),
             sd = ~sd(., na.rm = TRUE),
             min = ~min(., na.rm = TRUE),
             max = ~max(., na.rm = TRUE)
           ))
  ) %>%
  pivot_longer(everything(), names_to = "stat", values_to = "value")

# Create formatted table dynamically
stats_table <- tibble(
  Variable = labels,
  N = nrow(analysis_data),
  Mean = sapply(vars, function(v) mean(analysis_data[[v]], na.rm = TRUE)),
  SD = sapply(vars, function(v) sd(analysis_data[[v]], na.rm = TRUE)),
  Min = sapply(vars, function(v) min(analysis_data[[v]], na.rm = TRUE)),
  Max = sapply(vars, function(v) max(analysis_data[[v]], na.rm = TRUE))
)

# Save summary stats
write.csv(stats_table, file.path(analysis, "output", "tables", "summary_stats.csv"),
          row.names = FALSE)

# Summary by treatment status
summary_by_treatment <- analysis_data %>%
  group_by(treated) %>%
  summarise(
    n_obs = n(),
    mean_hr = mean(hr_fatalities),
    mean_nhr = mean(nhr_fatalities),
    mean_ln_hr = mean(ln_hr),
    mean_ln_nhr = mean(ln_nhr),
    .groups = "drop"
  )

write.csv(summary_by_treatment,
          file.path(analysis, "output", "tables", "summary_by_treatment.csv"),
          row.names = FALSE)

cat("  Summary statistics saved\n")
print(stats_table)

File: tutorial/worked-examples/bac-replication-python/analysis/code/01_summary_stats.py
# 01_summary_stats.py - Summary statistics

import pandas as pd
import numpy as np

# Load analysis data
analysis_data = pd.read_parquet(BUILD / "output" / "analysis_data.parquet")

print(f"  Observations: {len(analysis_data)}")
print(f"  States: {analysis_data['state_fips'].nunique()}")
print(f"  Years: {analysis_data['year'].min()}-{analysis_data['year'].max()}")

# Create output directories
(ANALYSIS / "output" / "tables").mkdir(parents=True, exist_ok=True)
(ANALYSIS / "output" / "figures").mkdir(parents=True, exist_ok=True)

# Summary statistics
summary_cols = ['total_fatalities', 'hr_fatalities', 'nhr_fatalities',
                'treated', 'ln_hr', 'ln_nhr']
if 'unemployment' in analysis_data.columns:
    summary_cols.append('unemployment')
if 'income' in analysis_data.columns:
    summary_cols.append('income')

summary_stats = analysis_data[summary_cols].describe().T
summary_stats = summary_stats[['count', 'mean', 'std', 'min', 'max']]

# Save
summary_stats.to_csv(ANALYSIS / "output" / "tables" / "summary_stats.csv")
print(f"  Saved summary statistics to analysis/output/tables/summary_stats.csv")

# Also by treatment status
by_treated = analysis_data.groupby('treated')[summary_cols].agg(['mean', 'std', 'count'])
by_treated.to_csv(ANALYSIS / "output" / "tables" / "summary_by_treatment.csv")

print("\n  Summary Statistics:")
print(summary_stats.round(2).to_string())
